Code-the-Dream-School/ee-prac-team3-front
# Feature/reset user progress loading bugfix

## Purpose:
To allow user to update their user progress and fix a bug with improper loading and rendering of quiz cards

## Summary:
Updated front end to:
- refactor multiple useEffect hooks in Main and Favorites components into a custom hook that allows sets a loading state to false only after all necessary data is present
- use backend "delete attempt" call to reset user data for individual quizzes (not for entire quiz platform).

## Front End Change:
App.js
Main.jsx
QuizCard.jsx
Favorites.jsx
exportFunctions.js

## Test Steps:
Reviewer may add any user progress to their account (once logged in) by completing a quiz.  Afterwords, the reviewer can click the three dots on the corner of a quiz with progress and choose "reset progress" and the progress should reset.
